kebechet/config.py

```diff
@@ -46,36 +48,72 @@ def from_file(self, config_path: str):
                 content = config_file.read()
 
         try:
-            self._repositories = yaml.safe_load(content).pop('repositories') or []
+            self._repositories = yaml.safe_load(content).pop("repositories") or []
         except Exception as exc:
-            raise ConfigurationError("Failed to parse configuration file: {str(exc)}") from exc
+            raise ConfigurationError(
+                "Failed to parse configuration file: {str(exc)}"
+            ) from exc
+
+    def download_conf_from_url(self, url: str, service: str):
+        if service not in services:
+            _LOGGER.warning("Service not supported")
+            raise ValueError(f"{service} is not supported at this time")
+
+        params = url.split("/")
+        slug = "/".join(params[3:])
+        if service == "gitlab":
+            slug = slug.replace("/", "%2F")
+        token = os.environ["GIT_ACCESS_TOKEN"]
+        _LOGGER.info(f"{service} service detected")
+        download_uri = services[service]["download_url"].format(slug=slug)
+        _LOGGER.info(f"Downloading from {download_uri}")
+        auth_value = services[service]["auth"]["value"].format(token=token)
+        auth_header = services[service]["auth"]["header"]
+        resp = requests.get(download_uri, headers={auth_header: auth_value})
+        file_ = tempfile.NamedTemporaryFile()
+        file_.write(resp.content)
+        return file_
```

Comment:
This all can be moved to `Service` class so you have full control there as described in my comment above. This manipulation seems to be too "low-level", hard to test and error prone.